packages/artifact/src/artifact.ts

```diff
@@ -0,0 +1,111 @@
+import * as core from '@actions/core'
+import {SearchResult, findFilesToUpload} from './search'
+import {
+  createArtifactInFileContainer,
+  uploadArtifactToFileContainer,
+  patchArtifactSize
+} from './upload-artifact-http-client'
+import {UploadInfo} from './upload-info'
+import {UploadOptions} from './upload-options'
+import {checkArtifactName} from './utils'
+
+/**
+ * Uploads an artifact
+ *
+ * @param name the name of the artifact, required
+ * @param path the directory, file, or glob pattern to denote what will be uploaded, required
+ * @param options extra options for customizing the upload behavior
+ * @returns single UploadInfo object
+ */
+export async function uploadArtifact(
+  name: string,
+  path: string,
+  options?: UploadOptions
+): Promise<UploadInfo> {
+  checkArtifactName(name)
+
+  if (!path) {
+    throw new Error('Upload path must be provided')
+  }
+
+  // Search for the items that will be uploaded
+  const filesToUpload: SearchResult[] = await findFilesToUpload(name, path)
+
+  if (filesToUpload === undefined) {
```

Comment:
`filesToUpload` should never be `undefined` now. Did some small updates. Initially I had that check because Typescript would complain that it could be undefined